go/cmd/vtorc/docgen/main.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/spf13/cobra"
+
+	"vitess.io/vitess/go/cmd/internal/docgen"
+	"vitess.io/vitess/go/cmd/vtorc/cli"
+)
+
+func main() {
+	var dir string
+	cmd := cobra.Command{
+		Use: "docgen [-d <dir>]",
```

Comment:
What is the current plan for documentation generation? This program seems very specific to only `vtorc`, should we have a top level generation tool that generates the docs for all binaries?

Reply:
each binary currently needs a `docgen` directory (primarly because you need to import _that_ command's root `cobra.Command` which is hard to parameterize). We can look into codegen with a template if it becomes too laborious, but right now I _think_ it's going to be a set-and-forget situation (write this directory once, then further changes to the actual binary don't require any changes here).

You can see how this is being used (also for `vtctldclient`) here: https://github.com/vitessio/website/pull/1576